src/include/sof/list.h

```diff
@@ -72,6 +72,30 @@ static inline void list_item_append(struct list_item *item,
 	list->prev = item;
 }
 
+/* adds item before reference item */
+static inline void list_item_add_before(struct list_item *ref_item,
```

Comment:
list_item_prepend() adds to the beginning of the list and list_item_append() at the end. These new functions allow to insert into specific location in the middle of the list.

Reply:
yes, but by changing the list head passed to list_item_prepend() should do the same.

Reply:
@lgirdwood 
Yes, you are right. If we replace list head, ```list_item_prepend()``` do the same as ```list_item_add_after()``` and ```list_item_append()``` do the same ```list_item_add_before()```. I can use "old" ones, but it can enter some misunderstanding by function names.

Reply:
@lgirdwood 
I've removed ```list_item_add_after()``` and ```list_item_add_before()``` definitions. I'm using ```list_item_append()``` instead of ```list_item_add_before()``` now.